### ISSUE TITLE
Detect interpolation in ''-enclosed strings

### ISSUE THREAD

**Issue by perlun** [bbatsov/rubocop#4324]
It's easy to make a typing mistake like this:

```ruby
foo = 'something with #{interpolation} inside'
```

when you intended to write

```ruby
foo = "something with #{interpolation} inside"
```

This is almost always an error. It would be great if Rubocop could spot that, and help me find it faster (i.e. before I run my code and notice that it's b0rken. 😆)

## RuboCop version

```
$ rubocop -V
0.48.1 (using Parser 2.4.0.0, running on ruby 2.4.1 x86_64-darwin16)
```

**Comment by bbatsov** [comment]
Agreed. That would be an useful lint cop as I guess.

**Comment by GauthamGoli** [comment]
Can I work on this? 
The way I see it,  `#{ }` in `str` nodes should raise the warning, where as `"` enclosed strings with interpolation would be `dstr` nodes. Is this the right way to do it?

**Comment by bbatsov** [comment]
> Can I work on this?

Sure.

> Is this the right way to do it?

Exactly.

**Comment by chrisseaton** [comment]
How does Rubocop think I should write strings that legitimately contain `#{something}`?
